### PR TITLE
EVG-6229, EVG-6238: stop agent monitor during host health checks in API server

### PR DESCRIPTION
**Pull request by Kimchelly**
JIRA:
https://jira.mongodb.org/browse/EVG-6229
https://jira.mongodb.org/browse/EVG-6238

* **EVG-6229**: If the API server does a host health check and it is not running, stop the host's agent monitor before sending response telling the agent to exit.
* **EVG-6238**: If host eventually enters the running state again. (e.g. static hosts are unquarantined), the agent monitor deploy job should automatically run because either `NeedsNewAgentMonitor` flag is set or `MaxLCTInterval` is elapsed.

### CODE REVIEW

File: model/host/hostutil.go [hunk 1]
```diff
@@ -511,14 +514,50 @@ func (h *Host) StartAgentMonitorRequest(settings *evergreen.Settings) (string, e
...
+	if err != nil {
+		return errors.Wrap(err, "could not get a Jasper client")
+	}
+
+	procs, err := client.Group(ctx, evergreen.AgentMonitorTag)
+	if err != nil {
+		return errors.Wrapf(err, "could not get processes with tag %s", evergreen.AgentMonitorTag)
+	}
+
+	grip.WarningWhen(len(procs) != 1, message.Fields{
```
Comment (john-m-liu) [inline model/host/hostutil.go]: This should probably log at error level

Comment (Kimchelly) [reply model/host/hostutil.go]: Done.

File: model/host/hostutil.go [hunk 2]
```diff
@@ -511,14 +514,50 @@ func (h *Host) StartAgentMonitorRequest(settings *evergreen.Settings) (string, e
...
+	if err != nil {
+		return errors.Wrap(err, "could not get a Jasper client")
+	}
+
+	procs, err := client.Group(ctx, evergreen.AgentMonitorTag)
+	if err != nil {
+		return errors.Wrapf(err, "could not get processes with tag %s", evergreen.AgentMonitorTag)
+	}
+
+	grip.ErrorWhen(len(procs) != 1, message.Fields{
```
Comment (tychoish) [inline model/host/hostutil.go]: probably a warning?

File: model/host/hostutil.go [hunk 3]
```diff
@@ -511,14 +514,50 @@ func (h *Host) StartAgentMonitorRequest(settings *evergreen.Settings) (string, e
...
+		"message": fmt.Sprintf("host should be running exactly one agent monitor, but found %d", len(procs)),
+		"host":    h.Id,
+		"distro":  h.Distro.Id,
+	})
+
+	catcher := grip.NewBasicCatcher()
+	for _, proc := range procs {
+		if proc.Running(ctx) {
+			catcher.Add(errors.Wrapf(proc.Signal(ctx, syscall.SIGTERM), "problem signalling process with ID %s", proc.ID()))
```
Comment (tychoish) [inline model/host/hostutil.go]: catcher has a wrapf method.

File: model/host/hostutil_test.go [hunk 4]
```diff
@@ -512,29 +512,32 @@ func TestJasperClient(t *testing.T) {
 	}
 }
 
+// withJasperServiceSetupAndTeardown performs necessary setup to start a
+// Jasper RPC service, executes the given test function, and cleans up.
+func withJasperServiceSetupAndTeardown(ctx context.Context, env *mock.Environment, manager *jasper.MockManager, h *Host, fn func()) error {
+	catcher := grip.NewBasicCatcher()
+	if !catcher.HasErrors() {
```
Comment (tychoish) [inline model/host/hostutil_test.go]: This will never be false.

I think it might be more clear to write this function without a catcher.

**Pull request merged**